src/test/librbd/deep_copy/test_mock_ObjectCopyRequest.cc

```diff
@@ -1104,5 +1104,80 @@ TEST_F(TestMockDeepCopyObjectCopyRequest, SkipSnapList) {
   ASSERT_EQ(0, ctx.wait());
 }
 
+TEST_F(TestMockDeepCopyObjectCopyRequest, RemoveObjectIncremental) {
```

Comment:
@trociny , is this sufficient or should we add more tests?

Reply:
It is good to have a mock test too, but I was talking about extending `src/test/librbd/test_DeepCopy.cc`. In addition to testing some specific cases it has stress tests that use a random factor that allowed us to catch bugs in the past when this feature was actively developed. As I said previously, unfortunately `test_DeepCopy.cc` tests only full image copy, and can't catch a bug like this, and I propose to extend it.

Reply:
@trociny , I have added  some tests to test_DeepCopy.cc.  Is this what you had in mind?
The tests currently fail and I am looking into that.

TestDeepCopy.SnapsIncremental was working earlier but is failing now because it cannot get an exclusive lock.

TestDeepCopy.CloneIncremental fails because of https://github.com/ceph/ceph/blob/d2205f5f8e13b7da2e2b40d5d68b372f49f6642b/src/librbd/io/ObjectRequest.cc#L891
As older snaps are not included, the deepcopy compute_read_ops fails to read all the required object data.

 I will continue to look into this.

Reply:
> TestDeepCopy.SnapsIncremental was working earlier but is failing now because it cannot get an exclusive lock.

The test succeeded earlier because it was run with RBD_FEATURES=1. It failed when RBD_FEATURE_EXCLUSIVE_LOCK was includes in RBD_FEATURES. 
 
@idryomov , thank you for figuring out that the exclusive_lock was not being acquired in the new API. TestDeepCopy.SnapsIncremental now passes.